### PR TITLE
Fixed Invalid date filter applied on FIM details flyout

### LINKED ISSUE #7146: Apply datetime filters on FIM details returns invalid date
**Description**

When the user clicks on the "Apply filter" on FIM details the filter pasted has an invalid date message 

**Steps to reproduce**
1. Navigate to FIM module
2. Click on a Files row
3. On the flyout details, clicks in the plus icon to a date filter

**Screenshots**

https://github.com/user-attachments/assets/c79b97c9-f2ca-43b9-9285-d700a357ae08

Screenshot 2024-11-12 at 11 56 08 AM

### PR DESCRIPTION
**Pull request by Machi3mfl**
### Description

This PR solves the problem when applying a date type filter by clicking on the fields of the FIM details flyout.
This problem only occurred in Firefox and Safari browsers.

Closes #7146

## Before

https://github.com/user-attachments/assets/c79b97c9-f2ca-43b9-9285-d700a357ae08

Screenshot 2024-11-12 at 11 56 08 AM

### Chrome

https://github.com/user-attachments/assets/17b90257-eb7c-462d-b0c7-3617071ec664

### Safari

https://github.com/user-attachments/assets/e3f725ce-2863-475a-bc21-b35a7afd2feb

### Firefox

https://github.com/user-attachments/assets/7a47238f-5eb3-42f3-8347-91f90e782684

## Test

1. Go to File Integrity Monitoring > Inventory Tab
2. Select an Agent
3. Click on the File table row
4. Apply the date filter by clicking the add filter button in the header time field
5. Repeat test for Safari, Chrome and Firefox browsers

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/public/components/agents/fim/inventory/fileDetail.tsx [hunk 1]
```diff
@@ -357,7 +358,12 @@ export class FileDetails extends Component {
                       >
                         <EuiButtonIcon
                           onClick={() => {
-                            this.addFilter(item.field, value);
+                            this.addFilter(
+                              item.field,
+                              item.field === 'date' || item.field === 'mtime'
+                                ? rawValue
+                                : value,
```
Comment (guidomodarelli) [inline plugins/main/public/components/agents/fim/inventory/fileDetail.tsx:365]: You have the option to create a new function called getValue, which can be designed to retrieve specific values as needed.

File: plugins/main/public/components/agents/fim/inventory/fileDetail.tsx [hunk 2]
```diff
@@ -357,7 +358,12 @@ export class FileDetails extends Component {
...
+                            );
```
Comment (guidomodarelli) [inline plugins/main/public/components/agents/fim/inventory/fileDetail.tsx:366]: You can create a new function called `addFilterHandler`

**Review by yenienserrano — APPROVED** [review]
LGTM

Safari

https://github.com/user-attachments/assets/f9b179c5-cb0c-41b4-9328-8679a02eb186

Firefox

https://github.com/user-attachments/assets/6cf8a05a-12c1-4517-a646-0361f6b5ccc8

**Review by asteriscos — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -43,6 +43,7 @@ All notable changes to the Wazuh app project will be documented in this file.
 - Fixed the Mitre ATT&CK exception in the agent view, the redirections of ID, Tactics, Dashboard Icon and Event Icon in the drop-down menu and the card not displaying information when the flyout was opened [#7116](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7116)
 - Fixed the filter are displayed cropped on screens of 575px to 767px in vulnerability detection module [#7047](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7047)
 - Fixed ability to filter from files inventory details flyout of File Integrity Monitoring [#7119](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7119)
+- Fixed Invalid date filter applied on FIM details flyout [#7160]()
```
[the remark this answers is not in the document]
Comment (Machi3mfl) [reply CHANGELOG.md]: Solved

**Review by asteriscos — DISMISSED** [review]
CR: :green_circle: 
Test: :green_circle: 

Firefox

Peek 2024-11-14 17-53

**Review by yenienserrano — DISMISSED** [review]

**Review by asteriscos — APPROVED** [review]

**Review by yenienserrano — APPROVED** [review]

**Pull request merged**